account/verification_code.go

```diff
@@ -0,0 +1,207 @@
+package account
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/fabric8-services/fabric8-auth/application/repository"
+	"github.com/fabric8-services/fabric8-auth/errors"
+	"github.com/fabric8-services/fabric8-auth/gormsupport"
+	"github.com/fabric8-services/fabric8-auth/log"
+
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+)
+
+type VerificationCode struct {
+	gormsupport.Lifecycle
+	ID       uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
+	User     User
+	Verified bool // The verification status of the updated email.
+	Code     string
```

Comment:
Let's maybe use ID as a code?

Reply:
We could and it would work, but didn't want to mix-up the entity ID with the what the entity contains 
:(

Reply:
OK. Let's stick with separate code field. We will have more flexible implementation and can change the code generation anytime (use some more complex/long strings for example).